### PR TITLE
Normalize documentation

### LINKED ISSUE #281: Clean up and normalize API documentation
It's _ok_ right now, but we should make sure every public function has consistent formatting and good docs.

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):**
- Fixes #281 

**Description:**
Bring all of the API documentation into line.

This includes API-breaking changes to bring the `StacIO` subclasses into agreement with their parent classes signatures, e.g. https://github.com/stac-utils/stactools/pull/315/files#diff-ccd61f3173dcc5f346076dd54f36876483c09bb8037b24269051e75225a1dad7R62. This PR _requires_ a major release (e.g. v0.4.0).

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/core/utils/antimeridian.py [hunk 1]
```diff
@@ -10,14 +11,17 @@
...
 class Strategy(Enum):
-    """Strategy for handling antimeridian-crossing polygons.
+    """Strategy for handling antimeridian-crossing polygons."""
 
-    - SPLIT: split the polygon at the antimeridian
-    - NORMALIZE: keep the polygon as one, but extend it to > 180 or < -180.
+    SPLIT = auto()
+    """Split the polygon into two or more polygons, such that none of them cross
+    the antimeridian.
     """
```
Comment (pjhartzell) [inline src/stactools/core/utils/antimeridian.py:19]: I've not seen this, where triple quotes are used underneath a definition. I'm guessing there is for some sphinx magic?

File: src/stactools/core/utils/antimeridian.py [hunk 2]
```diff
@@ -64,15 +68,18 @@ def split(polygon: Polygon) -> Optional[MultiPolygon]:
     If the polygon does not cross the antimeridian, returns None. Only handles
     exterior rings (can't handle interior).
 
-    NOTE: Will not work on polygons that enclose the north or south poles.
-    TODO: Fix this
+    Note:
+        Will not work on polygons that enclose the north or south poles.
+
+    Todo:
+        Fix this
```
Comment (pjhartzell) [inline src/stactools/core/utils/antimeridian.py:75]: I think I suggested including these back in the day. Feel free to clear out the Todo and whatever else.

**Review by pjhartzell — CHANGES_REQUESTED** [review]
Only concern is one place where it looks like merge conflict text made its way into a docstring. See comment.

File: src/stactools/core/io/__init__.py [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
 def read_text(href: str, read_href_modifier: Optional[ReadHrefModifier] = None) -> str:
+    """Reads a string from an href.
+
+    If ``read_href_modifier`` is provided, then ``href`` will be passed through
+    this function before use. This function uses the default
+    :py:class:`pystac.StacIO`.
+
+    Args:
+        href (str): The href to be read
+        read_href_modifier (ReadHrefModifier, optional): A function to modify
```
[the remark this answers is not in the document]
Comment (gadomski) [reply src/stactools/core/io/__init__.py:26]: When they start on the same line, they render as part of the argument (aka in bold) ... thanks for the catch, will fix.

File: src/stactools/core/utils/antimeridian.py (hunk 1 above)
Comment (gadomski) [reply src/stactools/core/utils/antimeridian.py:19]: Nope, just total habit for putting the main subject of the docs on the same line as the starting `"""` but wanting a newline before the last `"""`. It's a bastardization of https://peps.python.org/pep-0257/#multi-line-docstrings, which says that the first line should never wrap, but sometimes I wrap them for the sake of not spending a lot of brain cells on shortening the description. I'll shorten this one.

**Pull request merged**